### ISSUE TITLE
typescript/type-annotation-spacing incorrectly flagging union types

### ISSUE THREAD

**Issue by ismail-syed** [nzakas/eslint-plugin-typescript#106]
`typescript/type-annotation-spacing` is flagging the following code snippet: 

```
export interface Props {
  items: (string | number)[];
}
```

With the following message: 

```
error  Expected a space after the '('  typescript/type-annotation-spacing
```

I have the rule set to with default `'typescript/type-annotation-spacing': ['error']`. Is this a bug?

**Comment by macklinu** [comment]
If you change `['error']` to `'error'`, does ESLint report an error?

**Comment by macklinu** [comment]
The code snippet is valid in the test suite (diff below of the passing test):

```diff
diff --git a/tests/lib/rules/type-annotation-spacing.js b/tests/lib/rules/type-annotation-spacing.js
index ab26609..b5b47a8 100644
--- a/tests/lib/rules/type-annotation-spacing.js
+++ b/tests/lib/rules/type-annotation-spacing.js
@@ -22,6 +22,11 @@ const ruleTester = new RuleTester({
 ruleTester.run("type-annotation-spacing", rule, {
     valid: [
         `
+export interface Props {
+    items: (string | number)[];
+}
+        `,
+        `
 interface resolve {
     resolver: (() => PromiseLike<T>) | PromiseLike<T>;
 }

```

If this is still reproducible for you, could you please include your ESLint configuration as well? Thanks!

**Comment by ismail-syed** [comment]
I left `['error']` as it was before and it looks like this isn't flagging our code anymore. 
Strange :/
